cmd/notary/delegations.go

```diff
@@ -249,6 +220,48 @@ func (d *delegationCommander) delegationRemove(cmd *cobra.Command, args []string
 		}
 	}
 
+	delegationRemoveOutput(cmd, d, gun, role, keyIDs)
+
+	return maybeAutoPublish(cmd, d.autoPublish, gun, config, d.retriever)
+}
+
+func delegationAddInput(d *delegationCommander, cmd *cobra.Command, args []string) (
+	config *viper.Viper, gun string, role string, keyIDs []string, error error) {
```

Comment:
awesome!  I agree that it's worth taking a look at that refactor, thanks for opening the issue 👍